### ISSUE TITLE
[$250] [Report Fields] [External] Implement the policy report field edit flow

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#43685]
[Main tracking issue].(https://github.com/Expensify/Expensify/issues/365155)

[Design doc link](https://docs.google.com/document/d/10b7pTVvgpB6_O8sRTKaCzZEVM3Ba6hxnIwyyQSSXCLY/edit). 

Implement the following section of the design doc: https://docs.google.com/document/d/10b7pTVvgpB6_O8sRTKaCzZEVM3Ba6hxnIwyyQSSXCLY/edit#bookmark=id.mc44ht654nhy

This might technically be on hold for https://github.com/Expensify/App/issues/43677 but I think Callstack can start on these in parallel using feature branch.

Include unite tests for the optimistic actions

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a473de75c6065a37</li>
        <li>Upwork Job ID: 1801290900055702514</li>
        <li>Last Price Increase: 2024-06-13</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 102735611</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by waterim** [comment]
Hello, Im Artem from Callstack and would like to help with this issue

**Comment by rezkiy37** [comment]
Hi, I’m Michael (Mykhailo) from Callstack and I would like to work on this issue.

**Comment by sakluger** [comment]
Hey @rezkiy37 - is @waterim no longer working on this issue, or do we need multiple contributors for this one?

**Comment by rezkiy37** [comment]
Hey!
We need multiple contributors for this one.

**Comment by rezkiy37** [comment]
Actively working on the issue.

**Comment by rezkiy37** [comment]
Actively working on the issue.

**Comment by rezkiy37** [comment]
**Updates:**

1. Integrated the adding of list value.
2. Integrated the enabling/disabling of list value.

**Comment by rezkiy37** [comment]
I've opened the [PR](https://github.com/Expensify/App/pull/44730) for review 🙂

**Comment by rushatgabhane** [comment]
Approved and on production

**Comment by mountiny** [comment]
Can be closed now, thanks!
